networking_nsxv3/api/rpc.py

```diff
@@ -108,110 +81,77 @@ class NSXv3ServerRpcApi(object):
+    def get_qos_policies_with_revisions(self, limit, offset):
         cctxt = self.client.prepare()
-        return cctxt.call(self.context, 'get_qos_policy_revision_tuples',
-                          limit=limit, created_after=created_after)
+        return cctxt.call(self.context, 'get_qos_policies_with_revisions',
+                          host=self.host, limit=limit, offset=offset)
 
     @log_helpers.log_method_call
-    def get_security_group_revision(self, security_group_id):
+    def get_security_groups_with_revisions(self, limit, offset):
         cctxt = self.client.prepare()
-        return cctxt.call(self.context, 'get_security_group_revision',
-                          security_group_id=security_group_id)
+        return cctxt.call(self.context, 'get_security_groups_with_revisions',
+                          host=self.host, limit=limit, offset=offset)
 
     @log_helpers.log_method_call
-    def get_security_group_revision_tuples(
-            self, limit=_LIMIT, created_after=_CREATE_AFTER):
+    def get_security_group(self, security_group_id):
         cctxt = self.client.prepare()
-        return cctxt.call(self.context, 'get_security_group_revision_tuples',
-                          limit=limit, created_after=created_after)
-
-    @log_helpers.log_method_call
-    def has_security_group_tag(
-            self, security_group_id, tag_name):
-        cctxt = self.client.prepare()
-        return cctxt.call(self.context, 'has_security_group_tag',
-                          security_group_id=security_group_id,
-                          tag_name=tag_name)
+        return cctxt.call(self.context, 'get_security_group',
+                          security_group_id=security_group_id)
 
     @log_helpers.log_method_call
     def get_qos(self, qos_id):
         cctxt = self.client.prepare()
-        return cctxt.call(self.context, 'get_qos', qos_id=qos_id)
-
-    @log_helpers.log_method_call
-    def get_qos_bwl_rules(self, qos_id):
-        cctxt = self.client.prepare()
-        return cctxt.call(self.context, 'get_qos_bwl_rules', qos_id=qos_id)
-
-    def get_qos_dscp_rules(self, qos_id):
-        cctxt = self.client.prepare()
-        return cctxt.call(self.context, 'get_qos_dscp_rules', qos_id=qos_id)
+        return cctxt.call(self.context, 'get_qos', host=self.host, qos_id=qos_id)
 
     @log_helpers.log_method_call
     def get_port(self, port_id):
         cctxt = self.client.prepare()
-        return cctxt.call(self.context, 'get_port', port_id=port_id)
-
-    @log_helpers.log_method_call
-    def get_port_security_groups(self, port_id):
-        cctxt = self.client.prepare()
-        return cctxt.call(self.context, 'get_port_security_groups',
-                          port_id=port_id)
+        return cctxt.call(self.context, 'get_port', host=self.host, port_id=port_id)
 
     @log_helpers.log_method_call
-    def get_port_allowed_pairs(self, port_id):
+    def get_rules_for_security_groups_id(self, security_group_id):
```

Comment:
`get_rules_for_security_groups_id` should be `get_rules_for_security_group_id` (singular group id)

Reply:
- [x] Improve method naming to better reflect its return result